### PR TITLE
VSI + Dynamic Linking: setuid detection

### PR DESCRIPTION
**Pull request by kitified**
# Overview

Primitive to detect whether a file has `setuid` applied.

_I'm leaving this PR out of the changelog as it's literally irrelevant for users; I'll include this PR in a merged changelog message when I'm done (if anyone disagrees let me know!)_

## Acceptance criteria

Supports testing a file for the `setuid` bit.

## Testing plan

I relied on automated testing.

## Risks

None.

## References

Closes https://github.com/fossas/team-analysis/issues/884

## Checklist

- [x] I added tests for this PR's change (or confirmed tests are not viable).
- [x] If this PR introduced a user-visible change, I added documentation into `docs/`.
- [x] I updated `Changelog.md` if this change is externally facing. If this PR did not mark a release, I added my changes into an `# Unreleased` section at the top.
- [x] I updated `*schema.json` if I have made changes for `.fossa.yml`, `fossa-deps.{json, yaml, yml}`. You may also need to update these if you have added/removed new dependency (e.g. pip) or analysis target type (e.g. poetry).
- [x] I linked this PR to any referenced GitHub issues, if they exist.

### CODE REVIEW

**Review by meghfossa — APPROVED** [review]
LGTM 🚀

**Comment by kitified** [comment]
Heads up @meghfossa - made [a change](https://github.com/fossas/fossa-cli/pull/771/commits/4bff2e59cefbd1e140a268abfd44af801d5a09ba) after you approved, if you have opinions.

**Comment by kitified** [comment]
@meghfossa actually, that change wasn't a good one, because git strips the `setuid` bit. So I reverted it and added a nicer error message to the test. So no major change actually.

**Pull request merged**
